test/unit/type-registry-test.js

```diff
@@ -104,3 +106,25 @@ test('SolidProfile addTypeRegistry() test', function (t) {
 
   t.end()
 })
+
+test('type registry addToTypeIndex() updates the profile with new registry when there was previously no type index', t => {
+  nock('https://localhost:8443/')
```

Comment:
Nock is a nifty library for declaratively mocking out and responding to node's http requests.